### PR TITLE
Add determinant to Transform2D

### PR DESCRIPTION
**Pull request by Ughuuu**
Would be nice to have this function. It's also available and public in Godot, as can be seen here: [Transform2D](https://docs.godotengine.org/en/stable/classes/class_transform2d.html). Also, the function can be used to check for Transform2D invalid so that you don't call eg. affine_inverse on an invalid matrix and get an assert.

### CODE REVIEW

File: itest/rust/src/builtin_tests/geometry/transform2d_test.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ fn transform2d_equiv() {
 
     #[rustfmt::skip]
         let mappings_transform = [
+        ("determinant",      inner.determinant(),                                 outer.determinant()                                 ),
         ("affine_inverse",   inner.affine_inverse(),                             outer.affine_inverse()                             ),
         ("orthonormalized",  inner.orthonormalized(),                            outer.orthonormalized()                            ),
```
Comment (Bromeon) [inline itest/rust/src/builtin_tests/geometry/transform2d_test.rs:31]:
```suggestion
        ("determinant",      inner.determinant(),                                outer.determinant()                                ),
        ("affine_inverse",   inner.affine_inverse(),                             outer.affine_inverse()                             ),
        ("orthonormalized",  inner.orthonormalized(),                            outer.orthonormalized()                            ),
```

File: itest/rust/src/builtin_tests/geometry/transform2d_test.rs [hunk 2]
```diff
@@ -40,11 +41,6 @@ fn transform2d_equiv() {
         assert_eq_approx!(inner, outer, "function: {name}\n");
     }
 
-    assert_eq_approx!(
-        real::from_f64(inner.get_rotation()),
-        outer.rotation(),
-        "function: get_rotation\n"
-    );
     assert_eq_approx!(
```
Comment (Bromeon) [inline itest/rust/src/builtin_tests/geometry/transform2d_test.rs:44]: Why is this removed?

**Comment by GodotRust** [comment]
API docs are being generated and will be shortly available at: https://godot-rust.github.io/docs/gdext/pr-770

File: godot-core/src/builtin/transform2d.rs [hunk 3]
```diff
@@ -150,6 +150,18 @@ impl Transform2D {
...
 
+    /// Returns the determinant of the basis matrix. If the basis is uniformly
+    /// scaled, then its determinant equals the square of the scale factor.
+    /// 
+    /// A negative determinant means the basis was flipped, so one part of the
+    /// scale is negative. A zero determinant means the basis isn't invertible,
+    /// and is usually considered invalid.
+    ///
+    /// _Godot equivalent: `Transform2D.determinant()`_
+    pub fn determinant(&self) -> real {
```
Comment (Bromeon) [inline godot-core/src/builtin/transform2d.rs:161]: The 2nd part of the first sentence could be in the body (not header) of the docs, as it's not vital information.

You could also make the lines a bit wider, see also [here](https://godot-rust.github.io/book/contribute/conventions.html#formatting)  :slightly_smiling_face:

**Review by Bromeon — COMMENTED** [review]
Thanks a lot!

File: godot-core/src/builtin/transform2d.rs (hunk 3 above)
Comment (Ughuuu) [reply godot-core/src/builtin/transform2d.rs:161]: Hm, not sure I understand. You mean the second sentence? The first sentence is simply:

Returns the determinant of the basis matrix.

Should I leave this on it's own line? I tried to put this at the head (not sure what the header and the body is for docs, sorry)

Comment (Bromeon) [reply godot-core/src/builtin/transform2d.rs]: Sorry, I meant first paragraph. It's also not that important, don't spend too much time on this 🙂 

Typically, the structure is:
```
/// Describes what the function does (ideally more than the function name).
///
/// Describes detailed behavior and edge cases.
/// Can be longer text.
///
/// Even multiple paragraphs.
/// 
/// # Panics
/// Conditions that lead to panic (only if applicable -- not here).
/// 
/// # Safety
/// If the function is unsafe, what to consider (only if applicable -- not here).

**Comment by Ughuuu** [comment]
Enabled the actions on my fork to see how to fix the CI. will write here when I have it passing.

**Comment by Ughuuu** [comment]
Hm, not sure why the test fails. Any idea how to run it locally?

**Comment by lilizoey** [comment]
> Hm, not sure why the test fails. Any idea how to run it locally?

Since it's a 4.0 godot test failing, the best way i think is to:
1. Grab a 4.0 godot binary
2. Run `cargo build --p itest --features="godot/api-custom"` with `GODOT4_BIN` pointing to the 4.0 godot binary
3. Go into `itest/godot`
4. Run `godot4 --headless`

**Comment by Bromeon** [comment]
It's not necessary to run on Godot 4.0.

The error is very clear:
```rs
error[E0599]: no method named `determinant` found for struct `InnerTransform2D` in the current scope
  --> itest/rust/src/builtin_tests/geometry/transform2d_test.rs:44:30
   |
44 |         real::from_f64(inner.determinant()),
   |                              ^^^^^^^^^^^ method not found in `InnerTransform2D<'_>`
```

Then, navigate to [Godot 4 `Transform2D` docs](https://docs.godotengine.org/en/4.0/classes/class_transform2d.html) to confirm.
And you'll see that indeed, `determinant()` is missing. I has only been added later.

---

I'm fine to just add it for newer Godot versions. You can do this with `#[cfg(since_api = "4.1")]`, as the method is [available from 4.1](https://docs.godotengine.org/en/4.1/classes/class_transform2d.html).

Please also [squash your commits](https://github.com/godot-rust/gdext/blob/master/Contributing.md#one-commit-per-logical-change) and make sure there's a newline at the end of the `.gitignore`.

**Comment by Ughuuu** [comment]
Oh, so that was it, thanks, thats perfect. Didn't think it would be that simple, thought I had some configuration issue.

File: .gitignore [hunk 4]
```diff
@@ -23,3 +23,6 @@ gen
 *.bat
 config.toml
 exp.rs
+
+# Mac specific
+.DS_Store
```
Comment (fpdotmonkey) [inline .gitignore:28]: FYI you can add this to your `~/.gitconfig` to avoid needing to add this to every project's .gitignore
```
[core]
	excludesfile = /home/<USER>/.gitignore
```

Comment (Bromeon) [reply .gitignore:28]: I personally don't really mind this, as it's so common. If it achieves that not every single macOS contributor has to modify their global `.gitconfig`, then that's a win in my eyes.

**Review by Bromeon — APPROVED** [review]
Thanks!

**Comment by Bromeon** [comment]
Thanks!
